src/libmain/loggers.cc

```diff
@@ -21,7 +22,17 @@ LogFormat parseLogFormat(const std::string & logFormatStr) {
 }
 
 Logger * makeDefaultLogger() {
-    switch (defaultLogFormat) {
+    auto logFormat = defaultLogFormat;
+    std::string settingsLogFormat = settings.logFormat.get();
+    if (settingsLogFormat != "")
```

Comment:
If I'm not mistaken, this means that the `log-format` option takes precedence over the `log-format`.

You can probably get around that by changing the handler of the [log-format](https://github.com/NixOS/nix/blob/master/src/libmain/common-args.cc#L52-L58) option to `settings.set("log-format", format)` like it's done for example for [max-jobs](https://github.com/NixOS/nix/blob/master/src/libmain/common-args.cc#L60-L68)

Reply:
Quite right! I've done that.